Core_Method/CoreMethod_FWD/SDIRK_FWD_Integrator.m

```diff
@@ -62,14 +24,30 @@
 function [ Tout, Yout, ISTATUS, RSTATUS, Ierr, stack_ptr, quadrature ] = SDIRK_FWD_Integrator( OdeFunction,...
     Tspan, Y, OPTIONS, Coefficient, adjStackFlag, adjQuadFlag, stack_ptr )
 
+
+    % Force initial value matrix to be N X 1.
+    if ( size(Y,2) == 1 )
+        % DO NOTHING
+    else
+        Y = transpose(Y);
+    end  
+
     % Get Problem Size
     NVAR = max(size(Y));
 
     Tinitial = Tspan(1);
     Tfinal = Tspan(2);
+    
+   if ( OPTIONS.storeCheckpoint == true )
+        Yout = zeros(NVAR,OPTIONS.Max_no_steps);
+        Tout = zeros(OPTIONS.Max_no_steps,1);
+        TYindex = 1;
+        Yout(:,TYindex) = Y;
+        Tout(TYindex,1) = Tinitial; 
+    end   
```

Comment:
In all the other integrators we do adaptive reallocation of the matrices. The other integrators support Max_no_steps being infinite. This is one of the things I thought that I corrected. Maybe this is an older version of this file?